add dotnet list package topic

mairaw opened the pull request:
Contributes to #9256 

Replaces #9590  

Decided to do separate from PR #9395 to not mix updates with new content

mairaw committed:
add dotnet list package topic

zivkan commented on docs/core/tools/dotnet-list-package.md:
```diff
@@ -0,0 +1,101 @@
+
+  The NuGet sources to use when searching for newer packages. Requires the `--outdated` option.
+
+* **`--framework <FRAMEWORK | FRAMEWORK/RID>`**
```
We should check the latest 3.0 build to see if RID is still accepted. I know we did some work to remove RID from the output, since the packages should always be the same as the framework without the RID.

mairaw replied:
I have a work item to review the CLI as a whole for 3.0. Who would know about these changes?

rpetrusha replied:
If RIDs are accepted, is there a difference between \<FRAMEWORK | FRAMEWORK/RID>? Isn't \<FRAMEWORK/RID> adequate?

mairaw replied:
The format is FRAMEWORK/RID like `netstandard2.0/win-x64`

zivkan replied:
I had a quick chat with my team about it today. Previously (currently for GA versions) there are times when the SDK reports both TFM and TFM/RID to NuGet, causing NuGet to list the same packages list twice. This has been removed from the upcoming version, so NuGet will only report the TFM once. However, we didn't make any changes to the input parsing. So, while you can ask for `--framework netcoreapp2.0/win-x64`, it will either output netcoreapp2.0 packages without the win-x64 RID, or it won't output any packages at all. We need to add a work item to out backlog to remove the possibility of specifying RID as an input parameter. But for now the CLI allows you to specify `netcoreapp2.0/win-x64`, so I leave it up to your judgement if the docs should say this for now.

mairaw replied:
I think it's not an idea use-case, I'll just remove from the docs and keep just the framework option.

zivkan commented on docs/core/tools/dotnet-list-package.md:
````diff
@@ -0,0 +1,101 @@
+* List package references for a specific RID:
+
+  ```console
+  dotnet list package netstandard2.0/win-x64
````
Same as above, I'm not sure this is relevant with NuGet's latest changes.

mairaw replied:
I'm changing to a TFM example

zivkan commented on docs/core/tools/dotnet-list-package.md:
```diff
@@ -0,0 +1,101 @@
+
+The **Requested** column refers to the package version specified in the project file and can be a range. The **Resolved** column lists the version that the project is currently using and is always a single value. The packages displaying an `(A)` right next to their names represent [implicit package references](csproj.md#implicit-package-references) that are inferred from your project settings (`S ...
+
+Use the `--outdated` option to find out if there are newer versions available of the packages you're using on your projects. By default, `--outdated` lists the latest stable packages unless the resolved version is also a prerelease version. To include prerelease versions when listing newer versions, also specify the `--include-prerelease` option.
```
Would it be worthwhile to add a sample output when using the --outdated option?

zivkan replied:
yes, I think that would be interesting. If you're looking for a simple example, I think xunit is good. It's commonly used, has transitive dependencies, but not too many so the sample won't be too long.

zivkan reviewed (approved):
Thank you for creating these docs for us.

The information appears correct for the current version of the SDK, but we've got a change that will insert into the SDK at some point (we usually insert every few weeks) that will remove RIDs from the output. Without RIDs in the output, it doesn't make sense to allow it as an input, but we might need to change the CLI to enforce this.

rpetrusha commented on docs/core/tools/dotnet-list-package.md:
```diff
@@ -0,0 +1,101 @@
+
+## Description
+
+The `dotnet list package` command provides a convenient option to list all NuGet package references for a specific project or a solution. You first need to build the project in order to have the assets needed for this command to run. The following example shows the output of the `dotnet list package` command for the [SentimentAnalysis](https://github.com/dotnet/samples/tree/master/machine-learnin ...
```
```suggestion
The `dotnet list package` command provides a convenient option to list all NuGet package references for a specific project or a solution. You first need to build the project in order to have the assets needed for this command to process. The following example shows the output of the `dotnet list package` command for the [SentimentAnalysis](https://github.com/dotnet/samples/tree/master/machine-learning/tutorials/SentimentAnalysis) project:
```

rpetrusha commented on docs/core/tools/dotnet-list-package.md:
```diff
@@ -0,0 +1,101 @@
+
+The **Requested** column refers to the package version specified in the project file and can be a range. The **Resolved** column lists the version that the project is currently using and is always a single value. The packages displaying an `(A)` right next to their names represent [implicit package references](csproj.md#implicit-package-references) that are inferred from your project settings (`S ...
+
+Use the `--outdated` option to find out if there are newer versions available of the packages you're using on your projects. By default, `--outdated` lists the latest stable packages unless the resolved version is also a prerelease version. To include prerelease versions when listing newer versions, also specify the `--include-prerelease` option.
```
```suggestion
Use the `--outdated` option to find out if there are newer versions available of the packages you're using in your projects. By default, `--outdated` lists the latest stable packages unless the resolved version is also a prerelease version. To include prerelease versions when listing newer versions, also specify the `--include-prerelease` option.
```

rpetrusha commented on docs/core/tools/dotnet-list-package.md:
```diff
@@ -0,0 +1,101 @@
+
+* **`-h|--help`**
+
+  Prints out a short help for the command.
```
```suggestion
  Prints out short help for the command.
```

rpetrusha reviewed (approved):
I've left a few minor suggestions, @mairaw.

zivkan commented:
I noticed this hasn't been merged yet. We (NuGet team) fixed the issue of RIDs being displayed in the output (the output was always identical to the non-RID version, so it was useless duplicate information), although I noticed that we haven't updated the help text.

So, I suggest that these docs remove any information about RIDs and hopefully get merged. Thanks again for creating these docs for us. I wish we had someone to help us with nuget's own docs :)

nkolev92 commented on docs/core/tools/dotnet-list-package.md:
```diff
@@ -0,0 +1,102 @@
+
+* **`-s|--source <SOURCE>`**
+
+  The NuGet sources to use when searching for newer packages. Requires the `--outdated` option.
```
There's a new switch added in 3.0.1xx 
https://github.com/dotnet/cli/pull/10994

`--interactive`

Allows the command to stop and wait for user input or action (for example to complete authentication)

mairaw replied:
Will this be available with preview 4 @nkolev92? I can create an issue once that's out to add that flag.

nkolev92 replied:
Yeah, it will be.

mairaw commented:
@zivkan yes, I was planning to finish this one this week. I'll make sure to run with the latest tooling. Sorry about the delay 😢

mairaw committed:
add toc entry

mairaw committed:
updates

mairaw committed:
fix command

mairaw commented:
I believe I've addressed all the feedback. @rpetrusha can I get a new review of this PR?

The internal review URL is https://review.docs.microsoft.com/en-us/dotnet/core/tools/dotnet-list-package?branch=pr-en-us-10227
